brian2tools/baseexport/collector.py

```diff
@@ -412,3 +431,127 @@ def collect_PopulationRateMonitor(poprate_mon):
+
+    Returns
+    -------
+    synapse_dict : dict
+        Standard dictionary format with collected information
+    """
+    identifiers = set()
+    synapse_dict = {}
+    # get synapses object name
+    synapse_dict['name'] = synapses.name
+
+    # get source and target groups
+    synapse_dict['source'] = synapses.source.name
+    synapse_dict['target'] = synapses.target.name
+
+    # get governing equations
+    synapse_equations = collect_Equations(synapses.equations)
+    # get identifiers from equations
+    identifiers = identifiers | synapses.equations.identifiers
+    if synapses.event_driven:
+        synapse_equations.update(collect_Equations(synapses.event_driven))
+        identifiers = identifiers | synapses.event_driven.identifiers
+    # check equations is not empty
+    if synapse_equations:
+        synapse_dict['equations'] = synapse_equations
+    # check state updaters
+    if (synapses.state_updater and
+        isinstance(synapses.state_updater.method_choice, str)):
+        synapse_dict['user_method'] = synapses.state_updater.method_choice
+    # loop over the contained objects
+    summed_variables = []
+    pathways = []
+    for obj in synapses.contained_objects:
+        # check summed variables
+        if isinstance(obj, SummedVariableUpdater):
+            summed_var = {'code': obj.abstract_code, 'target': obj.target.name,
+                          'name': obj.name, 'dt': obj.clock.dt,
+                          'when': obj.when, 'order': obj.order
+                         }
+            summed_variables.append(summed_var)
+            # TODO: raises error because of _synaptic_var
+        # check synapse pathways
+        if isinstance(obj, SynapticPathway):
+            path = {'prepost': obj.prepost, 'event': obj.event,
+                    'code': obj.code, 'source': obj.source.name,
+                    'target': obj.target.name, 'name': obj.name,
+                    'clock': obj.clock.dt, 'order': obj.order,
+                    'when': obj.when
+                   }
+            # check delay is defined
+            if obj.variables['delay'].scalar:
+               path.update({'delay': obj.delay[:]})
+            pathways.append(path)
+            # check any identifiers specific to pathway expression
+            identifiers = identifiers | get_identifiers(obj.code)
+
+    # check any summed variables are used
+    if summed_variables:
+        synapse_dict['summed_variables'] = summed_variables
+    # check any pathways are defined
+    if pathways:
+        synapse_dict['pathways'] = pathways
+    # resolve identifiers and add to dict
+    identifiers = synapses.resolve_all(identifiers, run_namespace)
+    identifiers = _prepare_identifiers(identifiers)
+    if identifiers:
+        synapse_dict['identifiers'] = identifiers
+    # get when and order
+    # is this necessary or CodeRunner's is sufficient?
+    synapse_dict['when'] = synapses.when
+    synapse_dict['order'] = synapses.order
```

Comment:
The `when` is never used (I think), and the `order` is only used to set the `order` attribute of components. This is so that you can create a `Synapses(..., order=100)` for example and be sure that all its components will be executed later than another `Synapses(...)` object. But since your description contains `when`/`order` for all the components, no need to store this information here.

Reply:
Yes, thank you :) now it is clear. (I had that in my list of "to be cleared with you" and forgot)